ipu-plugin/pkg/ipuplugin/bridgeport.go

```diff
@@ -28,65 +28,82 @@ import (
 	"google.golang.org/protobuf/types/known/emptypb"
 )
 
-const (
-	outerVlanId = 0 // hardcoded s-tag
-)
-
+/*
+interfaces slice will be populated with PortIds on ACC(port representators->PRs on ACC, for Host VFs),
+for example, if ACC interface name is enp0s1f0d4, PortId(vportId) will be 4.
+Will also include port representators->PRs needed for Network Functions.
+*/
+var interfaces []uint
+var intfMap map[uint]bool
 var intfMapInit bool = false
 
-// Note: 3 reserved(last digit of interface name, for example, enp0s1f0d8, is 8) in exlude list in deviceplugin.
-var interfaces [3]string = [3]string{"6", "7", "8"}
-var intfMap map[string]bool
-
 func initMap() error {
-	if intfMapInit == false {
-		intfMap = make(map[string]bool)
+	var index uint
+	if !intfMapInit {
+		for index = HOST_VF_START_ID; index <= HOST_VF_END_ID; index = index + 1 {
+			interfaces = append(interfaces, index)
+		}
+		for index = NF_PR_START_ID; index <= NF_PR_END_ID; index = index + 1 {
+			interfaces = append(interfaces, index)
+		}
+		intfMap = make(map[uint]bool)
 		for _, intf := range interfaces {
 			intfMap[intf] = false
 		}
 		if len(interfaces) != len(intfMap) {
-			log.Errorf("initMap setup error\n")
-			return fmt.Errorf("initMap setup error\n")
+			log.Errorf("initMap setup error")
+			return fmt.Errorf("initMap setup error")
 		}
 		intfMapInit = true
 	}
 	return nil
 }
 
 // in-order(sorted by interface name->interfaces) allocation, based on available ACC interfaces(for Host VF)
-func allocateAccInterface() (error, string) {
-	var intfName string = ""
-	log.Debugf("allocateAccInterface\n")
-	if intfMapInit == false {
+func AllocateAccInterface(allocPr string) (uint, error) {
+	var intfId uint = 0
+	start, end := 0, 0
+
+	found := false
+	log.Debugf("AllocateAccInterface\n")
+	if !intfMapInit {
 		initMap()
 	}
-	for _, key := range interfaces {
-		log.Debugf("intfName->%v\n", key)
+	if allocPr == types.HostVfPr {
+		start = 0
+		end = HOST_VF_END_ID - HOST_VF_START_ID
+	} else {
+		start = HOST_VF_END_ID - HOST_VF_START_ID + 1
+		end = start + NF_PR_END_ID - NF_PR_START_ID
```

Comment:
This looks like here Host VF IDs range is from 0 to X and then NF ID range is from X to Y. However, in the const list in the other file, it seems like the NF ID range comes before the Host ID range?

Reply:
In the interfaces slice, the first 16 entries are used to store the host-VF interface IDs(that start from 22 to 37). And the rest of the slice is used to store interface IDs for NF PRs(from 6 to 13). Hope that clarifies.

Reply:
Maybe I am just dumb because I still don't understand. The first range is from 0 to (host_vf_end - host_vf_start). So 0 to (37-22) or; 0 to 16?
The second range is 17 to (17+13-6) so; 17 to 24 ?

Reply:
Maybe add a comment explaining how this maps to the static const values?

Reply:
Added comment to clarify.